src/routes/tools/local-climate-change-snapshot/_select-location/LocationForm.svelte

```diff
@@ -0,0 +1,242 @@
+<script>
+  import { Search } from "@berkeley-gif/cal-adapt-svelte-components";
+  import {
+    Button,
+    InlineNotification,
+    RadioButton,
+    RadioButtonGroup,
+  } from "carbon-components-svelte";
+  import { getTitle } from "~/helpers/geocode";
+  import { debounce } from "~/helpers/utilities";
+  import { geocodeSearch } from "./geocode-search";
+
+  /** the selected location's data as chosen by the user, this could also be set by the map */
+  export let selectedLocation = null;
+
+  /** the value of the search box */
+  export let searchValue = "";
+
+  /** the selected radio button boundary type */
+  export let selectedRadio = "address";
+
+  const buttonText = "Generate Snapshot".toUpperCase();
+  const inputDebounceMS = 300;
+  const minSearchLength = 3;
+  const searchLabelText = "Search for a place name or address";
+  const radioLegendText = "Select the type of location to search for";
+  const radios = [
+    {
+      label: "Address",
+      value: "address",
+    },
+    {
+      label: "County",
+      value: "counties",
+    },
+    {
+      label: "City",
+      value: "place",
+    },
+    {
+      label: "Census Tract",
+      value: "censustracts",
+    },
+    {
+      label: "Watershed (HUC10)",
+      value: "hydrounits",
+    },
+  ];
+
+  let searchSuggestions = [];
+  let abortController;
+  let notFound = false;
+  let showError = false;
+
+  $: isValid = selectedLocation !== null && !notFound;
+
+  $: if (searchValue === "") {
+    notFound = false;
+    selectedLocation = null;
+    searchSuggestions = [];
+  }
+
+  $: if (searchValue && searchValue.length && showError) {
+    showError = false;
+  }
+
+  $: {
+    console.log("searchSuggestions: ", searchSuggestions);
+    console.log("selectedLocation: ", selectedLocation);
+  }
+
+  function handleBtnClick() {
+    if (!searchValue) {
+      showError = true;
+    }
+  }
+
+  function handleRadioChange() {
+    handleAbortFetch();
+    searchValue = "";
+    searchSuggestions = [];
```

Comment:
probably don't need to set searchSuggestions and selectedLocation here, they should be set to the same value above when setting the searchValue to `""`.